pkg/virt-controller/watch/pool/pool.go

```diff
@@ -832,6 +839,55 @@ func (c *Controller) scale(pool *poolv1.VirtualMachinePool, vms []*virtv1.Virtua
 	return nil, false
 }
 
+func isVMIReady(vmi *virtv1.VirtualMachineInstance) bool {
+	if vmi.DeletionTimestamp != nil || vmi.Status.Phase != virtv1.Running {
+		return false
+	}
+	readyCondition := controller.NewVirtualMachineInstanceConditionManager().GetCondition(vmi, virtv1.VirtualMachineInstanceReady)
+	return readyCondition != nil && readyCondition.Status == k8score.ConditionTrue
```

Comment:
```suggestion
	return controller.NewVirtualMachineInstanceConditionManager().HasConditionWithStatus(vmi, virtv1.VirtualMachineInstanceReady, k8score.ConditionTrue)
```

Reply:
What you are doing in this helper is redundant. You can replace this helper with `controller.NewVirtualMachineInstanceConditionManager().HasConditionWithStatus(vmi, virtv1.VirtualMachineInstanceReady, k8score.ConditionTrue)` entirely.